modules/core/src/main/java/org/apache/ignite/internal/processors/service/GridServiceProxy.java

```diff
@@ -63,6 +69,22 @@
     /** */
     private static final long serialVersionUID = 0L;
 
+    /** */
+    private static final Method PLATFORM_SERVICE_INVOKE_METHOD;
+
+    static {
+        Method mtd;
+
+        try {
+            mtd = PlatformService.class.getMethod("invokeMethod", String.class, boolean.class, Object[].class);
+        }
+        catch (NoSuchMethodException e) {
```

Comment:
Let's either remove this `catch`, or throw a different exception explaining the problem. I don't think we should continue execution in this case.

Reply:
Unfortunatelly, we cannot remove this catch, because `NoSuchMethodException` this is checked one. You suggests to fail node? To be honest, I don't think that failing node in static initializer is a good variant. Let's explain. Let's imagine situatuin, that someone mistakenly delete `invokeMethod` from code, but then he or she get failing tests and easily correct this mistake.

Reply:
I suggest just to throw `ExceptionInInitializerError.html` [1](https://docs.oracle.com/javase/8/docs/api/java/lang/ExceptionInInitializerError.html), it's unchecked exception and created just for this case. What do you think?

Reply:
Fixed as proposed.

Reply:
Yes, we should fail the node, because this error indicates a nasty bug in our code.
Unchecked exception looks good to me.